pkg/services/ngalert/models/alert_rule.go

```diff
@@ -85,6 +85,10 @@ const (
 	// This isn't a hard-coded secret token, hence the nolint.
 	//nolint:gosec
 	ScreenshotTokenAnnotation = "__alertScreenshotToken__"
+
+	// These labels are Grafana controlled and non-overrideable. However, they are not the same as internal labels as
+	// they are meeant to be visible and passed-through to AMs. Should always be prefixed with "g_".
+	FolderUIDLabel = "g_folder"
```

Comment:
We need to make sure this prefix 
1. is documented
2. overrides the labels that come from the user. Probably by renaming them.